src/fullscreen/fullscreen-controller.js

```diff
@@ -195,7 +202,13 @@ class FullscreenController {
    */
   _requestExitFullscreen(): void {
     Promise.resolve(this._nativeExitFullScreen()).then(
-      () => (this._isInFullscreen = false),
+      () => {
+        this._isInFullscreen = false;
+        const screenOrientation = screen && screen.orientation && typeof screen.orientation.lock === 'function';
```

Comment:
shouldnt you check unlock here (although if lock exists it is same)?

Reply:
maybe it is enough to check for `screen && screen.orientation`?